intelmq/bots/experts/domain_valid/expert.py

```diff
@@ -0,0 +1,114 @@
+# -*- coding: utf-8 -*-
+"""
+Domain validator
+
+SPDX-FileCopyrightText: 2021 Marius Karotkis <marius.karotkis@gmail.com>
+SPDX-License-Identifier: AGPL-3.0-or-later
+"""
+
+import validators
+
+import os.path
+import pathlib
+import sys
+
+import requests.exceptions
+
+from intelmq.lib.bot import Bot
+from intelmq.lib.exceptions import MissingDependencyError
+from intelmq.lib.utils import get_bots_settings, create_request_session
+from intelmq.bin.intelmqctl import IntelMQController
+
+
+class DomainValidExpertBot(Bot):
+    domain_field: str = 'source.fqdn'
+    tlds_domains_list: str = '/opt/intelmq/var/lib/bots/domain_valid/tlds-alpha-by-domain.txt'
+
+    def init(self):
+        if validators is None:
+            raise MissingDependencyError("validators")
+        self.tlds_list = self.get_tlds_domain_list()
+
+    def process(self):
+        event = self.receive_message()
+        is_valid = False
+        if self.domain_field in event:
+            if validators.domain(event[self.domain_field]) and '_' not in event[self.domain_field] and \
+                    event[self.domain_field].split('.')[-1] in self.tlds_list:
+                is_valid = True
+            else:
+                self.logger.debug(f"Filtered out event with search field {self.domain_field!r}.")
+
+        if is_valid:
+            self.send_message(event)
+        self.acknowledge_message()
+
+    def get_tlds_domain_list(self):
+        if os.path.isfile(self.tlds_domains_list):
+            with open(self.tlds_domains_list) as file:
+                next(file)
+                lines = {line.strip().lower() for line in file}
+        else:
+            self.logger.debug(f"TLD domain list file not found at {self.tlds_domains_list!r}.")
```

Comment:
If the file is not found, that's an error as the bot would reject all domains a invalid. Or am I misunderstanding something?

Reply:
I think a better option to stop the bot